extensions/Worker.Extensions.ServiceBus/src/ServiceBusSessionMessageActionsConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Microsoft.Azure.ServiceBus.Grpc;
+using System.Text.Json;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind to <see cref="ServiceBusSessionMessageActions" /> or <see cref="ServiceBusSessionMessageActions[]" /> type parameters.
+    /// </summary>
+    [SupportsDeferredBinding]
+    [SupportedTargetType(typeof(ServiceBusSessionMessageActions))]
+    [SupportedTargetType(typeof(ServiceBusSessionMessageActions[]))]
+    internal class ServiceBusSessionMessageActionsConverter : IInputConverter
+    {
+        private readonly Settlement.SettlementClient _settlement;
+
+        public ServiceBusSessionMessageActionsConverter(Settlement.SettlementClient settlement)
+        {
+            _settlement = settlement;
+        }
+
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            try
+            {
+                var foundSessionId = context.FunctionContext.BindingContext.BindingData.TryGetValue("SessionId", out object? sessionId);
+                if (!foundSessionId)
+                {
+                    throw new InvalidOperationException("Expecting SessionId within binding data and value was not present.");
```

Comment:
Is the exception message here sufficient to help users understand what is wrong/how to fix the problem?

I believe the primary root cause for this would be attempting to bind to `ServiceBusSessionMessageActions`/`ServiceBusSessionMessageActions[]` when sessions are disabled, correct?

If that is the case, can we:

1. Improve the exception message to share that information. That would put the user facing the exception in a better position to fix the problem
2. Consider an analyzer. Wondering if we should consider an analyzer for this, so if a user is binding to a type that requires session, we detect whether the trigger is setting `IsSessionEnabled` to true. *this wouldn't have to be part of this PR, but let's open an issue*. Avoiding this at compilation time would be great.

Reply:
For the message, can we add something like: 

``` csharp
$"Sessions must be enabled when binding to {nameof(ServiceBusSessionMessageActions)}."
```

Reply:
Added it!